crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -753,15 +756,32 @@ impl<'db> TypeInferenceBuilder<'db> {
     }
 
     fn infer_augmented_assignment_statement(&mut self, assignment: &ast::StmtAugAssign) {
-        // TODO this should be a Definition
+        if assignment.target.is_name_expr() {
+            self.infer_definition(assignment);
+        } else {
+            // TODO currently we don't consider assignments to non-Names to be Definitions
+            self.infer_augment_assignment(assignment);
+        }
+    }
+
+    fn infer_augment_assignment_definition(
+        &mut self,
+        assignment: &ast::StmtAugAssign,
+        definition: Definition<'db>,
+    ) {
+        let target_ty = self.infer_augment_assignment(assignment);
+        self.types.definitions.insert(definition, target_ty);
```

Comment:
I think we also need to insert the type into `types.expressions` to avoid a similar issue to what I fixed in  https://github.com/astral-sh/ruff/pull/12919

Reply:
ah never mind, I think that works because `infer_expression` is already called below

Reply:
Yes, this wouldn't face the same issue.